messenger.go

```diff
@@ -48,9 +60,28 @@ func (m *messenger) Send(channel string, binaryMessage []byte) (binaryReply []by
 		return nil, errors.New("failed to send message")
 	}
 
-	// NOTE: Response from engine is not yet supported by embedder.
-	// https://github.com/flutter/flutter/issues/18852
-	return nil, nil
+	// wait for a reply and return
+	return <-reply, nil
+}
+
+// SendNoReply pushes a binary message on a channel to the Flutter side without
+// expecting replies.
+func (m *messenger) SendNoReply(channel string, binaryMessage []byte) (err error) {
```

Comment:
It's a bit inconsistent that the BinaryMessenger has Send and SendNoReply, but the channels have SendWithReply and Send (no reply). Perhaps change the BinaryMessenger to align with the latter?